debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java

```diff
@@ -342,7 +343,7 @@ private void createDataEventsForReplicaSet(ChangeEventSourceContext sourceContex
 
         LOGGER.info("Beginning snapshot of '{}' at {}", rsName, rsOffsetContext.getOffset());
 
-        final List<CollectionId> collections = primaryClient.collections();
+        final Set<CollectionId> collections = determineAllowedDataCollectionsForSnapshot(primaryClient.collections());
```

Comment:
There is a planned Jira for snapshotting collections based on the include list order. Could we keep this as a List?

Reply:
In methods like `io.debezium.relational.RelationalSnapshotChangeEventSource#determineCapturedTables` a Set is required. so i could one of two things:

- make `determineDataCollectionsToBeSnapshotted` to take a Collection<T> and return a `List<T>` and in the above method i can wrap it again in a set by doing `new HashSet<>(determineDataCollectionsToBeSnapshotted(getAllTableIds(ctx)))`

- what do you feel if we used a `LInkedHashSet` as it would maintain insertion order.

Reply:
@KaushikIyer16, sorry, I'm not quite following: how is this change for MongoDB related to `RelationalSnapshotChangeEventSource` (which only is used by the relational Debezium connectors)?

Reply:
the same method is being used to filter out the collection names in both relational and non-relational cases. in one case it deals with sets and the other with lists. hence, I settled down to Set